stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,175 @@
+     */
+    protected static void decompress(byte[] fileContentAsByteArray, String outputFolder) {
+        ZipInputStream zin = null;
+        try {
+            File outdir = new File(outputFolder);
+            zin = new ZipInputStream(new ByteArrayInputStream(fileContentAsByteArray));
+            ZipEntry entry;
+            String name, dir;
+            while ((entry = zin.getNextEntry()) != null) {
+                name = entry.getName();
+                if (entry.isDirectory()) {
+                    mkdirs(outdir, name);
+                    continue;
+                }
+        /* this part is necessary because file entry can come before
+         * directory entry where is file located
+         */
+                dir = dirpart(name);
+                if (dir != null) {
+                    mkdirs(outdir, dir);
+                }
+
+                extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            log.debug("I/O Exception when processing files ", e);
+            log.error("I/O Exception when processing files " + e.getMessage());
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                log.debug("I/O Exception when closing the input stream ", e);
+                log.error("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name
+     * @throws IOException
+     */
+    private static void extractFile(ZipInputStream in, File outdir, String name) {
+        byte[] buffer = new byte[4096];
+        BufferedOutputStream out = null;
+        try {
+            out = new BufferedOutputStream(new FileOutputStream(new File(outdir, name)));
+
+            int count = -1;
+            while ((count = in.read(buffer)) != -1) {
+                out.write(buffer, 0, count);
+            }
+        } catch (FileNotFoundException e) {
+            log.debug("File not found to process " + outdir, e);
+            log.error("File not found to process " + e.getMessage());
+        } catch (IOException e) {
+            log.debug("I/O Exception when closing the input stream " + outdir, e);
+            log.error("I/O Exception when closing the input stream " + e.getMessage());
+        } finally {
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    log.debug("I/O Exception when closing the input stream ", e);
+                    log.error("I/O Exception when closing the input stream " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create the directory name.
+     *
+     * @param name
```

Comment:
Add missing descriptions. Check all the methods

Reply:
Fixed with f97b66c